modules/participant_portal/php/survey_handle.class.inc

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * Survey_Handle
+ *
+ * PHP Version 7
+ *
+ * @category   Module
+ * @package    Main
+ * @subpackage Participant Portal
+ * @author     Shen Wang <shen.wang2@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris-Trunk/
+ */
+namespace LORIS\participant_portal;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+/**
+ * PHP Version 7
+ *
+ * @category   Module
+ * @package    Main
+ * @subpackage Participant_Portal
+ * @author     Shen Wang <shen.wang2@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris-Trunk/
+ */
+class Survey_Handle extends \NDB_Form
+{
+    public $skipTemplate = true;
+    /**
+     * GET method gets a list of Participant's categories and
+     * the children's categories.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        if ($request->getMethod()=='POST') {
+            $values = $request->getParsedBody();
+            $result = $this->_validate($values);
+            if ($result) {
+                return (new \LORIS\Http\Response())
+                    ->withHeader("Content-Type", "text/plain")
```

Comment:
The body of this (and the next one) seems to be JSON, not text/plain?

Is there a reason to not just use `new \LORIS\Http\Response\JSON\OK($result)`?